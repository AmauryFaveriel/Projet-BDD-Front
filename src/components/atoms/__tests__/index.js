import React from 'react';
import renderer from 'react-test-renderer';

// This is the only way I found to test our styled components without specific logic

// Import components
import BigTitle from '../big-title';
import MediumTitle from '../medium-title';
import NavItem from '../nav-item';
import RegularText from '../regular-text';
import SubTitle from '../subtitle';

describe('BigTitle', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BigTitle>Content</BigTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MediumTitle', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MediumTitle>Content</MediumTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('NavItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavItem>Content</NavItem>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('RegularText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RegularText>Content</RegularText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SubTitle', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubTitle>Content</SubTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
