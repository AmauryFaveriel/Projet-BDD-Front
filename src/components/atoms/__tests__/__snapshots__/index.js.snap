// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigTitle renders correctly 1`] = `
<label
  className="sc-AxjAm fSUMXK"
>
  Content
</label>
`;

exports[`MediumTitle renders correctly 1`] = `
<label
  className="sc-AxirZ UfKYF"
>
  Content
</label>
`;

exports[`NavItem renders correctly 1`] = `
<p
  className="sc-AxiKw esykbW"
>
  Content
</p>
`;

exports[`RegularText renders correctly 1`] = `
<label
  className="sc-AxhCb icaIA-D"
>
  Content
</label>
`;

exports[`SubTitle renders correctly 1`] = `
<label
  className="sc-AxhUy bmcXeL"
>
  Content
</label>
`;
